src/halmos/sevm.py

```diff
@@ -643,71 +743,98 @@ class Exec:  # an execution path
     code: Dict[Address, Contract]
     storage: Dict[Address, Dict[int, Any]]  # address -> { storage slot -> value }
     balance: Any  # address -> balance
+
     # block
     block: Block
+
     # tx
-    calldata: List[Byte]  # msg.data
-    callvalue: Word  # msg.value
-    caller: Address  # msg.sender
-    this: Address  # current account address
+    context: CallContext
+
     # vm state
+    this: Address  # current account address
     pgm: Contract
     pc: int
     st: State  # stack and memory
     jumpis: Dict[str, Dict[bool, int]]  # for loop detection
-    output: Any  # returndata
+    output: Bytes  # returndata
     symbolic: bool  # symbolic or concrete storage
     prank: Prank
+    addresses_to_delete: Set[Address]
+
     # path
     solver: Solver
     path: List[Any]  # path conditions
     alias: Dict[Address, Address]  # address aliases
-    # logs
-    log: List[Tuple[List[Word], Any]]  # event logs emitted
+
+    # internal bookkeeping
     cnts: Dict[str, Dict[int, int]]  # opcode -> frequency; counters
     sha3s: Dict[Word, int]  # sha3 hashes generated
     storages: Dict[Any, Any]  # storage updates
     balances: Dict[Any, Any]  # balance updates
     calls: List[Any]  # external calls
-    failed: bool
-    error: str
 
     def __init__(self, **kwargs) -> None:
         self.code = kwargs["code"]
         self.storage = kwargs["storage"]
         self.balance = kwargs["balance"]
         #
         self.block = kwargs["block"]
+        self.context = kwargs["context"]
         #
-        self.calldata = kwargs["calldata"]
-        self.callvalue = kwargs["callvalue"]
-        self.caller = kwargs["caller"]
         self.this = kwargs["this"]
-        #
         self.pgm = kwargs["pgm"]
         self.pc = kwargs["pc"]
         self.st = kwargs["st"]
         self.jumpis = kwargs["jumpis"]
-        self.output = kwargs["output"]
         self.symbolic = kwargs["symbolic"]
         self.prank = kwargs["prank"]
+        self.addresses_to_delete = kwargs.get("addresses_to_delete") or set()
         #
         self.solver = kwargs["solver"]
         self.path = kwargs["path"]
         self.alias = kwargs["alias"]
         #
-        self.log = kwargs["log"]
         self.cnts = kwargs["cnts"]
         self.sha3s = kwargs["sha3s"]
         self.storages = kwargs["storages"]
         self.balances = kwargs["balances"]
         self.calls = kwargs["calls"]
-        self.failed = kwargs["failed"]
-        self.error = kwargs["error"]
 
-        assert_address(self.caller)
+        assert_address(self.context.message.target)
```

Comment:
> we also need to assert for caller, not only for target, right?

right, good catch

> btw, can target be different from this?

I can't think of an example, it might be good a idea to get rid of `this` as a separate field